### PR TITLE
add api/topics endpoint

### PR DESCRIPTION
**Pull request by pzadev**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -22,3 +23,32 @@ describe("GET /api", () => {
...
+      {
+        description: "Not dogs",
+        slug: "cats",
+      },
+      {
+        description: "what books are made of",
+        slug: "paper",
+      },
+    ];
+
```
Comment (nc-gary-evans) [inline __tests__/app.test.js:43]: The data above does not need to be hardcoded, what tools are avialable to us so we can assert that everything is as it should be? for instance if your test data had 50 topics this would be a rather lengthy test suite if we declared a variable in each test this way.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -22,3 +23,32 @@ describe("GET /api", () => {
...
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual(topics);
+      });
+  });
+  test("should return 404 for invalid request", () => {
+    return request(app).get("/api/topics/1").expect(404);
+  });
+});
```
Comment (nc-gary-evans) [inline __tests__/app.test.js:54]: Where in your code have you implemented an error handler to send back a 404 status, and what other info should we send back to a user to accompany this status code?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -22,3 +23,32 @@ describe("GET /api", () => {
...
+        slug: "paper",
+      },
+    ];
+
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual(topics);
+      });
```
Comment (nc-gary-evans) [inline __tests__/app.test.js:49]: How can we let the user know what infomation the array contains? when sending a response you should be clear as to what the info is, have a look at your endpoints.json file to see what a response should look like.

File: app.js [hunk 4]
```diff
@@ -1,10 +1,29 @@
...
+// Error Handling
+
+app.use((err, req, res, next) => {
+  if (err.status === 400) {
+    return res.status(400).send({
+      error: "Bad Request",
+    });
+  }
+  next(err);
+});
```
Comment (nc-gary-evans) [inline app.js:19]: when implementing error handlers you should also implement a test before following TDD principles.

File: __tests__/app.test.js (hunk 3 above)
Comment (nc-gary-evans) [inline __tests__/app.test.js:49]: Also how can we test that we have received all of the topics and not just some? or none?
